config/src/main/kotlin/org/springframework/security/config/web/servlet/AuthorizeRequestsDsl.kt

```diff
@@ -152,12 +200,10 @@ class AuthorizeRequestsDsl : AbstractRequestMatcherDsl() {
                     is MatcherAuthorizationRule -> requests.requestMatchers(rule.matcher).access(rule.rule)
                     is PatternAuthorizationRule -> {
                         when (rule.patternType) {
-                            PatternType.ANT -> requests.antMatchers(rule.pattern).access(rule.rule)
-                            PatternType.MVC -> {
-                                val mvcMatchersAuthorizeUrl = requests.mvcMatchers(rule.pattern)
-                                rule.servletPath?.also { mvcMatchersAuthorizeUrl.servletPath(rule.servletPath) }
-                                mvcMatchersAuthorizeUrl.access(rule.rule)
-                            }
+                            PatternType.ANT -> requests.antMatchers(rule.httpMethod, rule.pattern).access(rule.rule)
+                            PatternType.MVC -> requests.mvcMatchers(rule.httpMethod, rule.pattern)
+                                .servletPath(rule.servletPath)
```

Comment:
I think we should keep the null check for `servletPath`.
I can see that we have null checks in `MvcRequestMatcher`, but I consider this scenario to be different than the `httpMethod`. 
We are never explicitly replacing an unset `servletPath` with null, as we did for `httpMethod`.

Reply:
I see. I removed it because:

1. The default value of `servletPath` on `MvcRequestMatcher` is `null` and the documentation for [`setServletPath()`](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/web/servlet/util/matcher/MvcRequestMatcher.html#setServletPath-java.lang.String-) states "The default is undefined which means any servlet path". This seems to be saying that `null` is equivalent to being unset, which means any servlet path.
1. In `PatternAuthorizationRule`, the default value is also `null` when unspecified. So we can take advantage of the above point to keep the code simpler here.

Would you still like me to add the null check back?

Reply:
I added it back, but I'm happy to take it out again if you prefer!